src/bgen/TypeManager.cs

```diff
@@ -152,149 +237,154 @@ public Type GetUnderlyingNullableType (Type type)
 		return type.GenericTypeArguments [0];
 	}
 
-	public void Initialize (BindingTouch binding_touch, Assembly api, Assembly corlib, Assembly platform)
+	public void Initialize (BindingTouch bindingTouch, Assembly api, Assembly corlib, Assembly platform)
```

Comment:
If you make this a constructor, you can avoid a lot of the nullable fields, since you'd be setting them in a constructor.

Reply:
@rolfbjarne  is not that easy, the time manager is used by the Generator, and making such a change would make me do a lot of changes they way in which the main method is deal with. I would not like to touch that code yet until we have tests for all this.